configurecompiler.cmake

```diff
@@ -464,8 +464,15 @@ if (CLR_CMAKE_PLATFORM_UNIX)
     add_compile_options(-Wno-unused-private-field)
     # Explicit constructor calls are not supported by clang (this->ClassName::ClassName())
     add_compile_options(-Wno-microsoft)
-    # This warning is caused by comparing 'this' to NULL
-    add_compile_options(-Wno-tautological-compare)
+
+    if (NOT COMPILER_HAS_ROBUST_TAUTOLOGICAL_COMPARISON_CHECKS)
```

Comment:
I do not think we want to be doing configure checks for warning levels. I do not think that the complexity is worth it.